kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/TestCaseExecutorTest.kt

```diff
@@ -288,4 +289,45 @@ class TestCaseExecutorTest : FunSpec() {
       then(listener).should().exitTestCase(argThat { description == Description.root("wibble") }, argThat { status == TestStatus.Failure })
     }
   }
+}
+
+class MultipleTestTimeoutTest : StringSpec() {
+
+  // The test executor was failing because as it reutilizes some threads from a thread pool.
+  // When using that thread pool, a task to cancel the thread is created, so that the engine can interrupt
+  // a test that is going forever.
+  // However, if the task
+
+  init {
+    // 100 millis sleep will "accumulate" between tests. If the context is still shared, one of them will fail
+    // due to timeout. This only happens on multi-thread, as on single-thread the interruption is syncronous, therefore
+    // it can't interrupt another thread in this spec
+    "Test 1".config(timeout = 200.milliseconds, threads = 2) {
```

Comment:
Probably a stupid question, but does `threads = 2` have any effect for a single test if `invocations` is still the default of `1`? I mean, if the test is only run once, what effect does `threads =2` have here?

Reply:
It doesn't. I added an extra thread because I know that internally, a thread pool will be used with 2+ threads, and the bug is on the thread pool. This is just me forcing it to use a thread pool

Reply:
Ok, thanks. As I believe that piece of information to be critical to understand the code, could you please try to make that more clear from your source code comment?

Reply:
Sure, makes sense